# Written and placed in public domain by Jeffrey Walton.
# This patch fixes some issues with Autogen.
--- config/guile.m4
+++ config/guile.m4
@@ -61,7 +61,7 @@
 #
 AC_DEFUN([GUILE_PKG],
  [PKG_PROG_PKG_CONFIG
-  _guile_versions_to_search="m4_default([$1], [2.0 1.8])"
+  _guile_versions_to_search="m4_default([$1], [2.2 2.0 1.8])"
   if test -n "$GUILE_EFFECTIVE_VERSION"; then
     _guile_tmp=""
     for v in $_guile_versions_to_search; do
--- configure
+++ configure
@@ -14191,7 +14191,7 @@
 		PKG_CONFIG=""
 	fi
 fi
-  _guile_versions_to_search="2.0 1.8"
+  _guile_versions_to_search="2.2 2.0 1.8"
   if test -n "$GUILE_EFFECTIVE_VERSION"; then
     _guile_tmp=""
     for v in $_guile_versions_to_search; do
--- agen5/guile-iface.h
+++ agen5/guile-iface.h
@@ -9,7 +9,7 @@
 # error AutoGen does not work with this version of Guile
   choke me.
 
-#elif GUILE_VERSION < 201000
+#elif GUILE_VERSION < 203000
 # define AG_SCM_IS_PROC(_p)           scm_is_true( scm_procedure_p(_p))
 # define AG_SCM_LIST_P(_l)            scm_is_true( scm_list_p(_l))
 # define AG_SCM_PAIR_P(_p)            scm_is_true( scm_pair_p(_p))
